service-providers/spotify-music-manager/main.py

```diff
@@ -17,6 +18,32 @@
 
 SEND_HEARTBEAT = True
 
+def getCommonFormatSongsFromPlaylists(userToken: str, playlistIDs: List[str]):
+    songs = []
+    for playlistID in playlistIDs:
+        playlist = api.getPlaylist(userToken, playlistID)
+        for item in playlist["items"]:
+            newSong = {}
+
+            newSong["songName"] = item["track"]["name"]
+            newSongArtists = []
+            newSongGenres = []
+            for artist in item["track"]["artists"]:
+                newSongArtists.append(artist["name"])
+                if ("genres" in artist):
+                    newSongGenres.extend(artist["genres"])
+            newSong["artists"] = newSongArtists
+            newSong["genres"] = newSongGenres
+            newSong["album"] = item["track"]["album"]["name"]
+            newSong["songLength"] = item["track"]["duration_ms"]
+            newSong["releaseDate"] = item["track"]["album"]["release_date"]
+            newSong["songId"] = item["track"]["id"]
```

Comment:
`"songId"` should be something like `"spotifySongId"` as we will need to know which service the ID is from for the importing part.

Reply:
importing will always ignore ids, so it doesn't really matter here. Good thought though 😄

Reply:
I just changed it to be spotifySongId and appleSongId, I'll just leave it as that